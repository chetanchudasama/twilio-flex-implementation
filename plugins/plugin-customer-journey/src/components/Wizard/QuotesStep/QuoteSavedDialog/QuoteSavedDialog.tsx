import React from "react";
import { ResponsiveDialog } from "@common/components";
import * as Flex from "@twilio/flex-ui";

export interface QuoteSavedDialogProps {
  open: boolean;
  handleQuoteSavedDialogClose: () => void;
}

const QuoteSavedDialog: React.FC<QuoteSavedDialogProps> = ({
  open,
  handleQuoteSavedDialogClose,
}) => {
  const handleBackToSearchCar = () => {
    handleQuoteSavedDialogClose();
    Flex.Actions.invokeAction("MoveToCarSearchStep", { tabIndex: 0 });
  };

  const handleViewAllQuotes = () => {
    handleQuoteSavedDialogClose();
    Flex.Actions.invokeAction("MoveToQuoteStep", {});
  };

  return (
    <ResponsiveDialog
      open={open}
      okText="View All Quotes"
      cancelText="Back To Car Search"
      isConfirmationDialog
      showActionButtons
      title="Quote saved!"
      onCancel={handleBackToSearchCar}
      onConfirm={handleViewAllQuotes}
    >
      This quote can be found on the quotes screen.
    </ResponsiveDialog>
  );
};

export default QuoteSavedDialog;
