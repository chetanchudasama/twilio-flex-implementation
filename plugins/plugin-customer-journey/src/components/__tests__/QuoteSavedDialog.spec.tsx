import React from "react";
import { create } from "react-test-renderer";
import QuoteSavedDialog, {
  QuoteSavedDialogProps,
} from "../Wizard/QuotesStep/QuoteSavedDialog/QuoteSavedDialog";

jest.mock("@common/components", () => {
  const components = jest.requireActual("@common/components");
  return {
    ...components,
    ResponsiveDialog: jest.fn(({ children }) => children),
  };
});

const props: QuoteSavedDialogProps = {
  open: true,
  handleQuoteSavedDialogClose: jest.fn(() => {}),
};

describe("QuoteSavedDialog", () => {
  it("renders correctly", async () => {
    const wrapper = create(<QuoteSavedDialog {...props} />).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
