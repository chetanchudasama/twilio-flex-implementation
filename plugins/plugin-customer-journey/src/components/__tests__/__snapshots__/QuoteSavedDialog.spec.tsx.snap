// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteSavedDialog renders correctly 1`] = `"This quote can be found on the quotes screen."`;
